Presentation/PresentationStyle.swift

```diff
@@ -89,53 +89,58 @@ public extension PresentationStyle {
             vc.modalPresentationCapturesStatusBarAppearance = capturesStatusBarAppearance ?? viewController.modalPresentationCapturesStatusBarAppearance
 
             return from.modallyPresentQueued(vc, options: options) {
-                Future { completion in
-                    let bag = DisposeBag()
-
-                    // The presentationController of an alert controller should not have its delegate modified
-                    if !(vc is UIAlertController) {
-                        /**
-                         Using a custom property instead of `viewController.presentationController?.delegate` because
-                         of a memory leak in UIKit when accessing the presentation controller of a view controller
-                         that's not going to be presented: https://github.com/iZettle/Presentation/pull/43#discussion_r307223478
-                         */
-                        let delegate = viewController.customAdaptivePresentationDelegate ?? CustomAdaptivePresentationDelegate()
-                        bag.hold(delegate)
-                        vc.presentationController?.delegate = delegate
-
-                        bag += delegate.shouldDismiss.set { presentationController -> Bool in
-                            guard !options.contains(.allowSwipeDismissAlways),
-                            let nc = (presentationController.presentedViewController as? UINavigationController) else {
-                                return true
-                            }
-                            return nc.viewControllers.count <= 1
-                        }
+                queuedModalPreparationFuture(for: vc, presentationController: vc.presentationController, options: options)
+            }
+        }
+    }
 
-                        bag += delegate.didDismissSignal.onValue { _ in
-                            completion(.failure(PresentError.dismissed))
-                        }
+    /// Creates a future to be used for queued modal presentations.
+    /// - Parameters:
+    ///   - viewController: The presented view controller.
+    ///   - presentationController: Optional custom presentation controller that will be used. Defaults to `nil`.
+    ///   - options: Presentation options.
+    static func queuedModalPreparationFuture(for viewController: UIViewController, presentationController: UIPresentationController? = nil, options: PresentationOptions) -> Future<()> {
```

Comment:
This method returns a future that can complete (in particular dismiss) a modal presentation and I think it should be called accordingly.
```suggestion
    static func modalPresentationDismissal(for viewController: UIViewController, presentationController: UIPresentationController? = nil, options: PresentationOptions) -> Future<()> {
```
Open for other suggestions too. The docs and usage will need to be updated after renaming.

Reply:
modalPresentationDismissalSetup?